client/client-multi/client-java-android/src/templates/log/DefaultLogCollector.java.template

```diff
@@ -38,12 +41,15 @@ public class DefaultLogCollector extends AbstractLogCollector {
     }
 
     @Override
-    public void addLogRecord(final ${log_record_class_package}.${log_record_class} record) {
+    public BucketFuture<BucketInfo> addLogRecord(final ${log_record_class_package}.${log_record_class} record) {
+        final BucketFuture<BucketInfo> future = new BucketFuture<>();
```

Comment:
By design, one future relates to one log bucket. According to this code snippet, a new future will be created for every log record. In the result, almost all future will never be completed.
Please, correct me if I am wrong.